rstudio/rstudio
# Factor out C++ specific options in `Options > Packages > C / C++`

### Intent

related to #10337

### Approach

<img width="654" alt="image" src="https://user-images.githubusercontent.com/2625526/149984184-95722186-e68a-4ebe-b454-0a7f013b7bcb.png">

### Automated Tests

> Indicate whether this change includes unit tests or integration tests, or link a work item or pull request to add those tests to another repo. If the change cannot or will not be covered by a test, indicate why.

### QA Notes

- Global options > Packages > C / C++ 
- select cpp11 as the template to use
- apply

- File > New File > New C++ file : this should create a file with: 

```cpp
#include <cpp11.hpp>

// This is a simple example of exporting a C++ function to R. You can
// source this function into an R session using cpp11::cpp_source()
// Learn more about cpp11 at: 
//
//   https://cpp11.r-lib.org/index.html

[[cpp11::register]]
double sum(cpp11::doubles x) {
    double total = 0.0;
    for (double value: x) {
        total += x;
    }
    return total;
}
```



### Checklist
